### ISSUE TITLE
GLUE evaluation numbers are very poor, if increase the sequence length to 512 and float 32

### ISSUE THREAD

**Issue by tbaggu** [JonasGeiping/cramming#28]
Hi 

I am trying to do some bench-marking as part of my experiments i want train BERT model with  512 sequence length and dtype as float 32 , i have pre trained the model wth above configuration and run the evaluation on glue_sne but the numbers are very poor.

May i know what went wrong

**Comment by JonasGeiping** [comment]
Hm, I haven't trained with that combination ever, hard to say what is going on. One sanity check question, during downstream finetuning, is the model also set to float32?

**Comment by tbaggu** [comment]
Yes,

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by JonasGeiping** [comment]
Hm just a note: There is also a separate `max_seq_length` setting in eval that is set only to 128 by default. Setting this to a lower number shouldn't make things much worse though

**Comment by tbaggu** [comment]
I have pre trained and fine turned for different  settings  and observed  steps vs loss for these settings, in case of 512 sequence length with fp32 ,I can see spikes in loss, I think it's due to learning rate, I have just reduced the lr, and re training, let me see how the results would be

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by JonasGeiping** [comment]
Closing this for now, cannot reproduce. Let me know if you find the source for this potential discrepancy.
